### PR TITLE
store heightmap data in a grid

### PR DESCRIPTION
**Pull request by antbern**
TLDR; Here is the PR mentioned in the discussion in #148 that changes the storage of the heightmap-like data to using an array along with metadata instead of as a XYZ point-cloud :rocket: 

Summary of changes:
- Store the heightmap-like data that was previously stored as binary XYZ files in a format more suitable for a grid in a `.hmap` file. Uses the `Vec2d` and stores it together with additional metadata like the `x` and `y` offsets as well as the `scale`.  This means we do not have to iterate all points to find min/max values etc and we can also directly index into the 2d-array instead of doing `HashMap` lookups (although in a few places I left the `HashMap` and just created it from the grid directly. This is to avoid having to go through those quite complicated algorithms in this PR. Instead here we just make sure we didn't break anything :smile: )
- To aid in reading and writing these binary formats, I decided to add a simple trait `FromToBytes` that can write `self` into a writer and construct `Self` from a reader. This makes the implementation quite simple and readable IMO :rocket: 
- The `Format` used to indicate whether the internal XYZ file contained point metadata or only points is no longer needed (it was used only for representing the height maps as XYZ files) , and has thus been removed. This simplifies the data format quite a bit.
- Command `internal2xyz` has been updated to support generating the old-style XYZ files from the new `.hmap` files.
- Went through `contours.rs` and figured that the `xyz2contours` function was actually performing _2 functions_. So I split it into two:  `xyz2heightmap` creates a `HeightMap` from a XYZ point cloud, `heightmap2contours` converts such a heightmap into `dxf` contours. This was also needed to do the change above and introduce the `HeightMap` type since the function was treating all input (either point cloud or heightmap-style as point cloud inputs).

Best of all: pipeline is still green which means the output _has not changed_ as a result of this. And I do believe we gained a few seconds in executions time! :100: 

(And yes I know the diff looks a bit crazy, especially for `contours.rs`. But it's because I split the `xyz2contours` into two which also changes the indentation... looking through the individual commits might in this case be easier :100: )

### CODE REVIEW

**Comment by antbern** [comment]
> Shaved 30sec from my 6min30sec job

Very nice! 💯
